client/interfaces.go

```diff
@@ -336,6 +336,9 @@ type ImageCreateArgs struct {
 
 	// Progress handler (called with upload progress)
 	ProgressHandler func(progress ioprogress.ProgressData)
+
+	// Type of the image (container or virtual-machine)
+	Type string
```

Comment:
@stgraber is it worth making this type `InstanceType` from the `api` package?

https://github.com/lxc/lxd/blob/master/shared/api/instance.go#L8-L17

This will then self document the allowed values.

Reply:
I'm not sure about that one yet. So far all the images type stuff has been done using strings, not using InstanceType. Part of the reason behind this was that, well, they're not instances.

I think it's probably best to go as is and if we decide that we should use InstanceType for images too, then we can do one change which converts everything in one go.

Reply:
@stgraber ah yeah that makes sense, maybe we could have some constants defining image types for now and then it would make finding them easier later (and avoiding typos).